web/apps/playground/src/components/PlaygroundApp.tsx

```diff
@@ -0,0 +1,265 @@
+          controlled
+          // @ts-ignore
+          autoCloseTags
+          smartIndent
+          detach
+          extensions={editorExtensions}
+          options={editorOptions}
+        />
+      </div>
+      {/* Divider for resizing (only when not collapsed) */}
+      {!isCollapsed && (
+        <div
+          className="h-2 cursor-row-resize bg-neutral-emphasis hover:bg-primary-border active:bg-primary-border transition-colors duration-100 z-10"
+          onMouseDown={handleMouseDown}
+          onDoubleClick={handleDividerDoubleClick}
+          role="separator"
+          aria-orientation="horizontal"
+          tabIndex={-1}
+        />
+      )}
+      {/* BottomPanel (Input/Output) */}
+      <div style={{ height: bottomPanelHeight, minHeight: COLLAPSED_PANEL_HEIGHT }}>
+        <BottomPanel isCollapsed={isCollapsed} setIsCollapsed={setIsCollapsed} />
+      </div>
+    </div>
+  );
+};
+
+export const PlaygroundApp = () => {
+  const setConfig = useSetAtom(configAtom);
+  const setLoading = useSetAtom(loadingAtom);
+  const setError = useSetAtom(errorAtom);
+  const setInterfaces = useSetAtom(interfacesAtom);
+  const [editorWidth, setEditorWidth] = useState(50); // percent
+  const dragging = useRef(false);
+  const bottomPanelRef = useRef<BottomPanelRef>(null);
+
+  useEffect(() => {
+    const params = getQueryParams();
+    const configParam = params.get("config");
+    const configUrl = params.get("configUrl");
+    setInterfaces(getInterfacesFromParams(params));
+
+    async function loadConfig() {
+      if (configParam) {
+        try {
+          // Parse url encoded config
+          // Replace all <br> tags with newlines
+          const decoded = decodeURIComponent(configParam).replace(/<br\s*\/?>/g, "\n");
+
+          setConfig(decoded);
+        } catch (e) {
+          setError("Failed to decode config. Are you sure it's a valid urlencoded string?");
+        }
+        return;
+      }
+      if (configUrl) {
+        setLoading(true);
+        try {
+          const res = await fetch(configUrl);
+          if (!res.ok) throw new Error("Failed to fetch config from URL.");
+          const text = await res.text();
+          // Replace all <br> tags with newlines
+          const decoded = text.replace(/<br\s*\/?>/g, "\n");
+          setConfig(decoded);
+        } catch (e) {
+          setError("Failed to fetch config from URL.");
+        } finally {
+          setLoading(false);
+        }
+      }
+    }
+    loadConfig();
+    // eslint-disable-next-line
+  }, [setConfig, setError, setLoading, setInterfaces]);
+
+  // Draggable divider logic
+  useEffect(() => {
+    const onMouseMove = (e: MouseEvent) => {
+      if (!dragging.current) return;
+      e.preventDefault();
+      document.body.style.cursor = "col-resize";
+      document.body.style.userSelect = "none";
+      const percent = (e.clientX / window.innerWidth) * 100;
+      setEditorWidth(Math.max(20, Math.min(80, percent)));
```

Comment:
can we set the default min and max (20 and 80) as constants so we're not using magic numbers